modules/data_models.py

```diff
@@ -0,0 +1,363 @@
+                                max_temperature_with_unit(str),
+                                min_temperature_with_unit(str)
+                            ]
+        """
+
+        if not self.days_data:
+            return None
+        month_data = []
+        for day_data in self.days_data:
+            month_data.append(
+                [
+                    self.year,
+                    self.name,
+                    day_data.get_date().split("-")[2],
+                    day_data.get_max_temperature(),
+                    day_data.get_min_temperature(),
+                ]
+            )
+
+        return month_data
+
+    def generate_averages_report_month(self):
+        """
+        Generates report using the averages list and displays
+        1. average highest temperature
+        2. average lowest temperature
+        3. average mean humidity
+        for a month
+        Returns:
+            None
+        """
+        averages = self.get_averages()
+        starting_message = ["Highest", "Lowest", "Mean Humidity"]
+        for index, val in enumerate(starting_message):
+            print(f"Average {val}: {averages[index]}")
+
+    def generate_report_charts(self):
+        """
+        This function displays month's report on console
+        A report looks like
+        "Month Year"
+        "day1 lowest_temp ++++++++++++++ highest_temp"
+        "day2 lowest_temp ++++++++++++++ highest_temp"
+        .
+        .
+        .
+        "day30 lowest_temp ++++++++++++++ highest_temp"
+        Returns:
+            None
+        """
+        extremes = self.get_month_values()
+        year, month = extremes[0][0:2]
+        print(f"{month} {year}")
+        for entry in extremes:
+            day = f"\33[0m{entry[2]}"
+            if entry[3] is None or entry[4] is None:
+                continue
+            red_plus = f"\33[91m{'+' * entry[3]}"
+            blue_plus = f"\33[94m{'+' * entry[4]}"
+            report_line = (
+                f"{day} {blue_plus}{red_plus} "
+                f"\33[0m{entry[4]}{TEMPERATURE_UNIT}-{entry[3]}{TEMPERATURE_UNIT}"
+            )
+            print(report_line)
+
+
+class YearData:
+    """
+    This class holds data of an entire year
+    """
+
+    def __init__(self, year, path=WEATHER_FILES_DIR):
+        """
+        Initializes the members year, path and calls populate to fill months_data list
+        Args:
+            year(str or int):   Value containing 4 digit year e.g '2004'
+            path(str):  Value containing path to weather files e.g 'weatherfiles/'
+        """
+        self.months_data = []
+        self.year = year
+        self.path = path
+        self.populate()
+
+    def populate(self):
+        """
+        Initializes the months_data member with MonthData objects
+        Returns:
+            None
+        """
+        for month in range(1, 13):
+            month_data = MonthData(self.year, get_month_name(month)[0:3], self.path)
```

Comment:
We shouldn't pass the formatted month name to this function, we can just pass the month number and it will be the responsibility of MonthData Class to parse and save it in a formatted way. You can probably create the function `get_name` in MonthData class to return the month name value based on that object's month number.

Reply:
Done